calus.js

```diff
@@ -1,228 +1,185 @@
                     }
 
-                    for (let day = monthStart; day <= monthEnd; day = day.plus({ days: 1 })) {
-                        while(availableDatesIndex < available.length && day > available[availableDatesIndex]) {
-                            availableDatesIndex++
-                        }
+                    let isAvailable = false
 
-                        let isAvailable = false
-                        if (availableDatesIndex < available.length && available[availableDatesIndex].hasSame(day, 'day')) {
-                            isAvailable = true
-                        }
+                    if (available.length && available[0].hasSame(day, 'day')) {
+                        isAvailable = true
+                    }
 
+                    let startOf = day.startOf('day')
 
-                        let startOf = day.startOf('day')
+                    days.push({
+                        time: day,
+                        isToday: day.hasSame(this.now, 'day'),
+                        isAvailable: isAvailable,
+                        isPast: startOf < startOfToday,
+                        isFuture: startOf > startOfToday,
+                        isThisMonth: day.hasSame(date, 'month'),
+                    })
+                }
 
-                        days.push({
-                            time: day,
-                            isToday: day.hasSame(this.now, 'day'),
-                            isAvailable: isAvailable,
-                            isPast: startOf < startOfToday,
-                            isFuture: startOf > startOfToday,
-                            isThisMonth: day.hasSame(date, 'month'),
-                        })
-                    }
+                const sameMonth = date.startOf('month').hasSame(this.now, 'month');
+                const allowScroll = !this.allowPrevScroll && sameMonth;
 
-                    months.push({
-                        time: date.startOf('month'),
-                        isCurrentMonth: +days[10].time.startOf('month') == +startOfToday.startOf('month'),
-                        isInCurrentYear: +days[10].time.startOf('year') == +startOfToday.startOf('year'),
-                        days: days
-                    })
+                months.push({
+                    time: date.startOf('month'),
+                    isCurrentMonth: +days[10].time.startOf('month') == +startOfToday.startOf('month'),
+                    isInCurrentYear: +days[10].time.startOf('year') == +startOfToday.startOf('year'),
+                    days: days,
+                    disablePrevScroll: allowScroll
+                })
 
-                    date = date.plus({ months: 1 })
-                }
+                date = date.plus({ months: 1 })
+            }
 
-                return months
+            return months
+        }
+    },
+    methods: {
+        // select a day
+        select: function (day) {
+            if (day.isAvailable) {
```

Comment:
As stated above, mark this as Deprecated (`console.warn()`) as handled via properties

Reply:
As above - not deprecated